contracts/base/ModuleManager.sol

```diff
@@ -40,6 +40,44 @@ abstract contract ModuleManager is SelfAuthorized, Executor, GuardManager {
         }
     }
 
+    /**
+     * @notice Runs pre-execution checks for module transactions if a guard is enabled.
+     * @param to Target address of module transaction.
+     * @param value Ether value of module transaction.
+     * @param data Data payload of module transaction.
+     * @param operation Operation type of module transaction.
+     * @param guard Guard to be used for checking.
+     * @return guardHash Hash returned from the guard tx check.
+     */
+    function runPreExecutionChecks(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        address guard
+    ) internal returns (bytes32 guardHash) {
```

Comment:
You can actually return the `guard` and `guardHash` from here, effectivly moving the `getGuard` call to this method.

Reply:
Nice idea, done 👍🏾